Migrate notice creation view/client logic to engine.

eschlange opened the pull request:
* Moved in toast notice dependencies into engine.
* Updated controller signatures to enable/disable notices.
* Updated controller logic to use new flag.
* Updated manifests for new dependencies.

[#95080280]

ericcf commented on app/views/social_networking/goals/tool.html.erb:
```diff
@@ -107,4 +107,6 @@
   angular.module('socialNetworking.values').value('participantStudyEndDate', '<%= current_participant.active_membership_end_date %>');
   angular.module('socialNetworking.values').value('currentGoals', <%= goals.to_json.html_safe %>);
+  angular.module('socialNetworking.values').value('noticesEnabled', <%= Rails.configuration.enable_notices %>);
+  angular.module('socialNetworking.values').value('noticeUtility', Notice);
```
Are you using `Notice` in other places? Is that why you're injecting it as a global variable?

ericcf commented on app/assets/javascripts/notices.js:
```diff
@@ -0,0 +1,86 @@
+var Notice = (function() {
```
Where are you testing this logic?

eschlange replied:
Yes, I have yet to check it in. I'm still working on a few tests for the udpates to the angular controllers.

eschlange commented:
@ericcf Hi Eric, I updated the JS code to follow Lint standards and added a few Javascript test cases for the Notices utility. Once this is merged, I'll integrate this into the SunnySide host application.

ericcf commented:
Thanks!
